core/json/src/main/scala/net/liftweb/json/Extraction.scala

```diff
@@ -265,11 +265,14 @@ object Extraction {
           val instance = jconstructor.newInstance(args.map(_.asInstanceOf[AnyRef]).toArray: _*)
           setFields(instance.asInstanceOf[AnyRef], json, jconstructor)
         } catch {
-          case e @ (_:IllegalArgumentException | _:InstantiationException) =>
-            fail("Parsed JSON values do not match with class constructor\nargs=" +
-                 args.mkString(",") + "\narg types=" + args.map(a => if (a != null)
-                   a.asInstanceOf[AnyRef].getClass.getName else "null").mkString(",") +
-                 "\nconstructor=" + jconstructor)
+          case exception: Exception =>
```

Comment:
Hmm… This went from a targeted catch to catching all exceptions and swallowing any exception that isn't an `IllegalArgumentException` or an `InstantiationException`, right?

Should we add a spec or two to confirm this is doing what we think it should be? For this particular one I think it's when the class that you're deserializing is either throwing an instantiation exception or an illegal argument exception.